components/camel-ai/camel-langchain4j-web-search/src/main/java/org/apache/camel/component/langchain4j/web/search/LangChain4jWebSearchConfiguration.java

```diff
@@ -0,0 +1,199 @@
+        this.resultType = resultType;
+    }
+
+    /**
+     * The {@link #maxResults} is the expected number of results to be found if the search request were made. Each
+     * search engine may have a different limit for the maximum number of results that can be returned.
+     *
+     * @return
+     */
+    public Integer getMaxResults() {
+        return maxResults;
+    }
+
+    public void setMaxResults(Integer maxResults) {
+        this.maxResults = maxResults;
+    }
+
+    /**
+     * The {@link #language} is the desired language for search results is a string that indicates that the search
+     * client desires search results in the specified language. Each search engine may have a different set of supported
+     * languages.
+     *
+     * @return
+     */
+    public String getLanguage() {
+        return language;
+    }
+
+    public void setLanguage(String language) {
+        this.language = language;
+    }
+
+    public String getGeoLocation() {
+        return geoLocation;
+    }
+
+    /**
+     * The {@link #geoLocation} is the desired geolocation for search results is a string that indicates that the search
+     * client desires search results in the specified geolocation. Each search engine may have a different set of
+     * supported geolocations.
+     */
+    public void setGeoLocation(String geoLocation) {
+        this.geoLocation = geoLocation;
+    }
+
+    public Integer getStartPage() {
+        return startPage;
+    }
+
+    /**
+     * The {@link #startPage} is the start page number for search results is the page number of the set of search
+     * results desired by the search user
+     */
+    public void setStartPage(Integer startPage) {
+        this.startPage = startPage;
+    }
+
+    public Integer getStartIndex() {
+        return startIndex;
+    }
+
+    /**
+     * The {@link #startIndex} is the start index for search results is the index of the first search result desired by
+     * the search user. Each search engine may have a different set of supported start indexes in combination with the
+     * start page number.
+     */
+    public void setStartIndex(Integer startIndex) {
+        this.startIndex = startIndex;
+    }
+
+    public Boolean getSafeSearch() {
+        return safeSearch;
+    }
+
+    /**
+     * The {@link #safeSearch} is the safe search flag is a boolean that indicates that the search client desires search
+     * results with safe search enabled or disabled.
+     */
+    public void setSafeSearch(Boolean safeSearch) {
+        this.safeSearch = safeSearch;
+    }
+
+    public Map<String, Object> getAdditionalParams() {
+        return additionalParams;
+    }
+
+    /**
+     * The {@link #additionalParams} is the additional parameters for the search request are a map of key-value pairs
+     * that represent additional parameters for the search request. It's a way to be flex and add custom param for each
```

Comment:
I think we should use `flexible` which is an adjective instead of `flex` which is a verb

Reply:
My bad, I copied text from Langchain4j javadoc. I had improved the text a bit. Let me know if that looks good to you.